addons/website_slides/wizard/slide_channel_invite.py

```diff
@@ -19,34 +19,50 @@ class SlideChannelInvite(models.TransientModel):
 
     # composer content
     attachment_ids = fields.Many2many('ir.attachment', string='Attachments')
+    send_email = fields.Boolean('Send Email')
     # recipients
     partner_ids = fields.Many2many('res.partner', string='Recipients')
     # slide channel
-    channel_id = fields.Many2one('slide.channel', string='Slide channel', required=True)
+    channel_id = fields.Many2one('slide.channel', string='Course', required=True)
+    channel_invite_url = fields.Char('Course Link', compute='_compute_channel_invite_url')
+    channel_visibility = fields.Selection(related='channel_id.visibility')
+    channel_published = fields.Boolean(related='channel_id.is_published')
+    # membership
+    with_enrollment = fields.Boolean(
+        'Enroll partners', readonly=True,
+        help='Whether invited partners will be added as enrolled. Otherwise, they will be added as invited.')
+
+    @api.depends('channel_id')
+    def _compute_channel_invite_url(self):
+        for invite in self:
+            channel = invite.channel_id
+            invite.channel_invite_url = f'{channel.get_base_url()}/slides/{channel.id}/invite'
+
+    @api.onchange('channel_visibility')
+    def _onchange_channel_visibility(self):
+        self.send_email = self.channel_visibility != 'public' or self.with_enrollment
 
     # Overrides of mail.composer.mixin
     @api.depends('channel_id')  # fake trigger otherwise not computed in new mode
     def _compute_render_model(self):
         self.render_model = 'slide.channel.partner'
 
-    @api.onchange('partner_ids')
-    def _onchange_partner_ids(self):
-        if self.partner_ids:
-            signup_allowed = self.env['res.users'].sudo()._get_signup_invitation_scope() == 'b2c'
-            if not signup_allowed:
-                invalid_partners = self.env['res.partner'].search([
-                    ('user_ids', '=', False),
-                    ('id', 'in', self.partner_ids.ids)
-                ])
-                if invalid_partners:
-                    raise UserError(_(
-                        'The following recipients have no user account: %s. You should create user accounts for them or allow external sign up in configuration.',
-                        ', '.join(invalid_partners.mapped('name'))
-                    ))
+    @api.model_create_multi
+    def create(self, vals_list):
+        for values in vals_list:
+            if values.get('template_id') and not (values.get('body') or values.get('subject')):
```

Comment:
Why this ? Normally the 'mail.compose.mixin' handles everything, subject and body are computed based on template.